workers/api-server/src/http.rs

```diff
@@ -390,6 +394,15 @@ impl HttpServer {
             GetOrderHistoryHandler::new(state.clone()),
         );
 
+        // --- External Match Routes --- //
+
+        // The "/external-match/request" route
+        router.add_unauthenticated_route(
```

Comment:
Certainly out of scope for this PR, but I am curious the auth method we'll use for this endpoint. We probably don't want to use a single symmetric secret like the other endpoints so that we can have better control over per-user access granting / revocation, right?

Reply:
yeah we'll allow relayers to issue api keys